services/src/main/java/org/exoplatform/addons/gamification/service/setting/zone/impl/ZoneRegistryImpl.java

```diff
@@ -39,15 +46,41 @@ public boolean remove(ZoneConfig zone) {
     @Override
     public void start() {
 
-        for (ZoneConfig zone : zoneMap.values()) {
-            CommonsUtils.getService(SettingService.class).set(Context.GLOBAL, Scope.GLOBAL.id(null), (GAMIFICATION_PREFERENCES + zone.getZoneName()), SettingValue.create(zone.getZoneName()));
+        try {
+            // Processing registered domains
 
+            for (ZoneConfig domain : zoneMap.values()) {
+                DomainDTO domainDTO = domainService.findDomainByTitle(domain.getZoneName());
+                if (domainDTO == null) {
+                    store(domain);
+                }
+            }
+        } catch (Exception e) {
+            LOG.error("Error when processing Domains ", e);
         }
 
+
     }
 
     @Override
     public void stop() {
 
     }
+
+    private void store(ZoneConfig zoneConfig) {
+
+        DomainDTO domainDTO = new DomainDTO();
+        domainDTO.setTitle(zoneConfig.getZoneName());
+        domainDTO.setDescription(zoneConfig.getZoneDescription());
+        domainDTO.setLastModifiedDate(LocalDate.now().toString());
+        domainDTO.setLastModifiedBy("Gamification");
+        domainDTO.setCreatedBy("Gamification");
+        domainDTO.setCreatedDate(LocalDate.now().toString());
+        try {
+            domainDTO.setPriority(Integer.parseInt(zoneConfig.getZonePriority()));
+        } catch (Exception e) {
+            domainDTO.setPriority(10);
```

Comment:
Does any action should/can be taken if a prority is wrong ? we should track the error to help the diagnosis

Reply:
The zone priority is useless property and not used for the moment, so wrong priority don't affect anything

Reply:
yes sure, nut at least you should log a warning to notice there is something wrong somewhere